### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Scriptkiddi**

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -1,41 +1,46 @@
 ---
-- name: Update repositories cache and install "php" package
+- name: Update repositories cache and install "php-fpm" package
   apt:
-    name: php-fpm
+    name: "{{ item }}"
     update_cache: yes
+  with_items:
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: why loop if you only install one package

File: tasks/main.yml [hunk 2]
```diff
@@ -1,41 +1,46 @@
...
-    patterns: "*" 
-    file_type: directory
-  register: php_version
+- name: delete default php-fpm pool.conf
+  file:
+    path: "/etc/php/{{ php_version }}/fpm/pool.d/www.conf"
+    state: absent
 
-- name: Update php.ini
+- name: "Update php.ini"
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: quotes are not necessary

File: templates/php-fpm.conf.j2 [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# {{ ansible_managed }}
+;# {{ ansible_managed }}
```
Comment (SlothOfAnarchy) [inline templates/php-fpm.conf.j2]: remove #

File: templates/pool.conf.j2 [hunk 4]
```diff
@@ -1,20 +1,21 @@
-# {{ ansible_managed }}
+;# {{ ansible_managed }}
```
Comment (SlothOfAnarchy) [inline templates/pool.conf.j2]: remove #

File: templates/pool.conf.j2 [hunk 5]
```diff
@@ -81,15 +83,13 @@ pm = {{ pool.pm | mandatory }}
...
+pm.max_children = {{ pool.pm_max_children | mandatory }}
 
 ; The number of child processes created on startup.
 ; Used only when pm is set to dynamic.
 ; Default Value: min_spare_servers + (max_spare_servers - min_spare_servers) / 2.
 {% if pool.pm.start_servers is defined %}
-    pm.start_servers = {{- pool.pm.start_servers -}}
-{% else %}    
-    pm.start_servers = {{- pm_start_servers  -}}
+    pm.start_servers = {{ pool.pm.start_servers }}
```
Comment (SlothOfAnarchy) [inline templates/pool.conf.j2]: please describe these vars in readme. The user can't even find them in the defaults.

File: templates/pool.conf.j2 [hunk 6]
```diff
@@ -186,10 +187,10 @@ pm.max_requests = {{ pool.pm.max_requests | default(pm_max_requests)}}
 
 ; The log file for slow requests. 
 ; Default value: #INSTALL_PREFIX#/log/php-fpm.log.slow.
-{% if pool.slowlog %}
-    slowlog = {{- pool.slowlog -}}
+{% if pool.slowlog is defined %}
```
Comment (SlothOfAnarchy) [inline templates/pool.conf.j2]: why are using `is defined` all of the sudden? this use case should have worked before

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -42,7 +42,8 @@ pools:
     pm: static
     pm.max_children = 20
 php_ini_values:
-  apc.enabled: 1
+  apc.enabled:
+    - 1
```
Comment (SlothOfAnarchy) [inline README.md]: why is this a list?

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

File: templates/pool.conf.j2 (hunk 5 above)
Comment (Scriptkiddi) [reply templates/pool.conf.j2]: read the php_fpm doc for this

File: README.md (hunk 7 above)
Comment (Scriptkiddi) [reply README.md]: because depending on what var you are setting multiple values are allowed and are merged by php into one var

Comment (SlothOfAnarchy) [reply README.md]: but this is not the case with `apc.enabled`?

File: README.md [hunk 8]
```diff
@@ -40,10 +40,15 @@ pools:
     listen: /path/to/unix/socket
     user: www-data
     pm: static
```
Comment (SlothOfAnarchy) [inline README.md:42]: are you sure this works? you assign a string to `pm`, but then treat it as a dict

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

**Review by SlothOfAnarchy — APPROVED** [review]

**Pull request merged**